### PR TITLE
Set up deployer tests in CI/CD

### LINKED ISSUE #106: Running deployer tests in Github Actions
If we have a test namespace set up - we can just run the deployer tests here. Need to set Github Actions to only run one test workflow at once to prevent clashes.

### PR DESCRIPTION
**Pull request by jthomas**
Will close #106. I've decided to move the test framework to be run via NPM rather than brew as this was the easiest solution to installing and running the test library + deps across different platforms. I need to set up the login credentials for the test namespace in the Github Actions secrets store to check this works.

I've also added NPM module caching to the CI/CD pipline to hopefully speed up the test runs....

### CODE REVIEW

**Comment by jthomas** [comment]
I've got this running now 🎊 - see results at https://github.com/jthomas/nimbella-cli/runs/2685012864?check_suite_focus=true

As secrets aren't available by default for the PR runner event type (to stop secrets being leaked...) - need to work out when to run these tests in the main repo. Probably just on commit and maybe nightly? Or we could try to work out how to set up the workflow to check if the user opening the PR is a member of Nimbella org and allow the test step to run if this is true.

File: package.json [hunk 1]
```diff
@@ -205,11 +185,12 @@
     "prepare": "npm run version-check && patch-package && husky install",
     "postpack": "rimraf oclif.manifest.json",
     "prepack": "rimraf lib && tsc -b && npm run minify && oclif-dev manifest && oclif-dev readme",
-    "test": "echo NO TESTS",
+    "test": "npx bats -r tests/*/*bats",
```
Comment (rabbah) [inline package.json:188]: nice!

**Review by rabbah — APPROVED** [review]

**Comment by rabbah** [comment]
Needs a rebase to resolve package-lock conflicts.

**Comment by rabbah** [comment]
> need to work out when to run these tests in the main repo. Probably just on commit and maybe nightly?

path of least resistance to start.

**Pull request merged**